src/components/Geogebra/GeogebraViews.js

```diff
@@ -74,15 +75,43 @@ export default class {
             }
         });
 
+        api.service('elements').on('removed', (element) => {
+            const pos = this.workshopIds.indexOf(element.workshop);
+            if (pos !== -1) {
+                this.GAVs[pos].deleteObject(element.name);
+            }
+        });
+
         api.service('workshops').on('xml-changed', (workshop) => {
             const pos = this.workshopIds.indexOf(workshop.id);
             if (pos !== -1) {
                 this.GAVs[pos].setXML(workshop.xml);
             }
         });
+    }
 
-        api.service('workshops').on('created', (workshop) => {
-            console.log('Workshop created', workshop);
-        });
+    mergeViews(workshopIds, params) {
+        this.mergedView = new GeogebraMergedAdminView({
+            container: 'merged_ggb_applet',
+            id: 'merged_ggb_applet',
+            perspective: 'G',
+            showResetIcon: false,
+            ...params,
+            log: this.log,
+        }, workshopIds);
+
+        this.mergedView.inject();
+    }
+
+    mergeGoLive() {
+        this.log.debug('Going live');
+        this.mergedView.clear();
+        this.mergedView.loadWorkshops();
```

Comment:
Shouldnt `initializeCallbacks` be before loading workhops? What will happen in an unlikely scenario that workshops will arrive before callbacks are set?

Reply:
You may be right, it may probably omit some events in-between